### DISCUSSION TITLE
Turborepo should not enforce setting experimental packageManager

### DISCUSSION THREAD

**Discussion by fabio-colella-md** [vercel/turborepo#8792] [Ideas]
### Goals

1. `packageManager` is an experimental property and should not be enforced
> Stability: 1 - Experimental. The feature is not subject to [semantic versioning](https://semver.org/) rules. Non-backward compatible changes or removal may occur in any future release. Use of the feature is not recommended in production environments.
Experimental features are subdivided into stages:

### Non-goals

1. Nothing else than goal 1

### Background

[`packageManager` is a property](https://nodejs.org/api/packages.html#packagemanager) that can optionally be added to a package.json to mandate the version of the package manager to use. This property, as well as [Corepack](https://nodejs.org/api/corepack.html#supported-package-managers) in general is experimental and not recommended for production. However, this property is now [required by turborepo since version 2](https://turbo.build/repo/docs/crafting-your-repository/upgrading#upgrading-to-20).

### Proposal

My proposal is to not enforce this rule in stable releases of turborepo until the feature itself is stable.

**Comment by fabio-colella-md** [comment]
As a tradeoff, one might set a similar property in the turbo.json file in the root, allowing similar flexibility as for packages, including `yarn@*` kind of patterns 😊

**Comment by anthonyshew** [comment]
This configuration option will be a part of 2.0.10: https://github.com/vercel/turbo/releases/tag/v2.0.10-canary.2

**Reply by fabio-colella-md** [reply]
Wonderful, thanks 🙏 Is this the PR you're referring to? https://github.com/vercel/turbo/pull/8738 😊

**Reply by anthonyshew** [reply]
Yes!
